hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/aws_sdk_v2_changelog.md

```diff
@@ -93,6 +93,20 @@ the bucket. The behaviour has now been changed to:
   response. Otherwise it will throw an error with status code 301 permanently moved. This error
   contains the region of the bucket in its header, which we can then use to configure the client.
 
```

Comment:
A lot of what is in this doc is outdated, it was written when we raised the initial PR for the upgrade to help with reviewing. For example, we don't do the HEAD call to get the region of the bucket. We had to do this when crossRegion was not supported in SDK V2. 

Let's create a new section in `index.md` which explains region handling, something like:

* `fs.s3a.endpoint` and `fs.s3a.endpoint.region` can be used to set values for endpoint and region. 
* If a value is set in `fs.s3a.endpoint.region` , S3A will configure the S3 Client to use this value. If this is set to a region that does not match your bucket, you will receive a 301 redirect response.
* If no region is set, but an endpoint is set, S3A will attempt to parse your region from the endpoint. 
* If your endpoint is set to the central `s3.amazonaws.com`, S3A will enable cross region access. This means that even if the region `fs.s3a.endpoint.region` is incorrect, the SDK will determine the region. This is done by making the request, and if the SDK receives a 301 redirect, it determines the region at the cost of a HEAD request, and caches it. 
* If no region is no set, and none could be determined from the endpoint, S3A will use US_EAST_2 as the default region and enable cross region access. Again, this means that requests will be not fail, but an initial HEAD request will be made to get the region of your bucket. 
* If the configured region is an empty string, we fallback to using the SDK region [resolution chain](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/region-selection.html).

I don't think we need to explain specifically what the code is doing in too much detail, rather just the end behaviour the user will see.

Reply:
* put it in `connecting.md`, link in index.md as "how to connect"
* Highlight that it is complex and improving, so may be considered unstable.
* Link to `third_party_stores.html` to make clear if you are working with third party stores to look there.